src/TraceEvent/AutomatedAnalysis/AnalyzerIssue.cs

```diff
@@ -10,11 +10,14 @@ public class AnalyzerIssue
         /// <summary>
         /// Create a new instance of AnalyzerIssue.
         /// </summary>
+        /// <param name="id">A durable id, in form of GUID, to uniquely identify a performance issue. Note: This id is intended to be stable for a given issue.
+        /// Once the analyzer decides the id for a specific issue, it does not change.</param>
         /// <param name="title">A string title.</param>
         /// <param name="description">A string description.</param>
         /// <param name="url">A URL pointing to further documentation.</param>
         /// <exception cref="InvalidOperationException">Instances of AnalyzerIssue can only be created during Analyzer execution.</exception>
         public AnalyzerIssue(
+            Guid id,
```

Comment:
Follow-up question @xiaomi7732 - should we force a value for the ID (e.g. throw if `id` is `Guid.Empty`)?  My gut reaction is yes we should force this.

Reply:
Also, this change is going to break the `Automated CPU Analysis` view in PerfView.  We should address this before merge.

Reply:
Hi Brian,
I can add a Guid.Empty check easily.
Could you please point me to the code for `Automated CPU Analysis`? And where do I invoke it on PerfView UI? I can't seem to locate it in this repo.

Reply:
It's located under the `Experimental` group of views.  This only shows up on corpnet today, but I am going to change that to be configurable in the PerfView UI.  If you're building PerfView, you can comment out the `InternalUser` check here:

https://github.com/microsoft/perfview/blob/main/src/PerfView/PerfViewData.cs#L7545

Reply:
Also, here is the exception that fires with your change, which is expected since the signature changed:
<img width="1387" alt="image" src="https://user-images.githubusercontent.com/6210322/219468976-a4428db2-6443-4470-bd5d-427fa7acd27a.png">

Reply:
@brianrob Things starts to make sense a bit. The exception is expected since the signature of the ctor changed. We will need a newer package for the analyzers to make it work.

But what is the flow?

How can I sync source between this repo and the repo with the analyzers? The analyzers wouldn't compile until it knows the new signature of the constructor of the analyzer issue. 

Or do you want me to directly work on the other repo, so that we can release a newer package for the analyzers, and then bump up the reference to the analyzer package here?

An alternative could be, instead of changing the ctor, there could be an overload, but I want to avoid it since we control both package versions.

Reply:
I have been doing some work recently here and have made some other changes that will break in the same way.  I'll go ahead and fix the rules along with my changes.